mitodl/micromasters
# Add script to use webpack dev server to serve javascript bundles

#### What are the relevant tickets?
Fixes #3232 

#### What's this PR do?
Adds a new script `scripts/test/run_selenium_tests_dev.sh` which runs the selenium tests using an existing webpack dev server instance. The user is required to set an environment variable `WEBPACK_DEV_SERVER_HOST` to tell selenium where to find the webpack dev server. The restriction that `DEBUG` is true is removed because tests will always have `DEBUG == False` and there is already `USE_WEBPACK_DEV_SERVER`

#### How should this be manually tested?
Set the environment variable `USE_WEBPACK_DEV_SERVER` to the IP address of the dev server (don't include the port number). This should generally be the same as whatever IP address you use to access your local dev environment (I think). Then run `./scripts/tests/run_selenium_tests_dev.sh`. All tests should pass.

Go to `CourseAction` and remove or modify the `enroll-button` CSS class from the `SpinnerButton`. Run the tests again and the `test_zero_price_purchase` test should fail.
